app/controllers/agents_controller.rb

```diff
@@ -184,9 +193,11 @@ def update
   end
 
   def leave_scenario
-    @agent = current_user.agents.find(params[:id])
-    @scenario = current_user.scenarios.find(params[:scenario_id])
-    @agent.scenarios.destroy(@scenario)
+    Agent.transaction do
```

Comment:
Hmm do we need this? Afaik we don't have agents that change the scenario of another Agent.